monitoring-dashboard/components/org.wso2.ei.dashboard.core/src/main/java/org/wso2/ei/dashboard/core/rest/delegates/heartbeat/HeartBeatDelegate.java

```diff
@@ -82,63 +85,81 @@ private boolean isNodeRegistered(HeartbeatObject heartbeat) {
+                }
+            };
+            heartbeatScheduledExecutorService.scheduleWithFixedDelay(runnableTask, 3 *
+                    heartbeatInterval, 3 * heartbeatInterval, TimeUnit.SECONDS);
+            return dataManager.insertHeartbeat(heartbeat, accessToken);
+        } catch (DashboardServerException e) {
+            throw new ManagementApiException (e.getMessage(), 500, e.getCause());
+        }
     }
 
     private boolean isNodeShutDown(HeartbeatObject heartbeat, String initialTimestamp) {
         return !dataManager.checkIfTimestampExceedsInitial(heartbeat, initialTimestamp);
     }
 
     private void deleteNode(String productName, HeartbeatObject heartbeat) {
-        int rowCount = dataManager.deleteHeartbeat(heartbeat);
-        if (rowCount > 0) {
-            logger.info("Successfully deleted node where group_id : " + heartbeat.getGroupId() + " and node_id : "
-                     + heartbeat.getNodeId() + ".");
-            deleteAllNodeData(productName, heartbeat);
-        } else {
+        try {
+            int rowCount = dataManager.deleteHeartbeat(heartbeat);
+            if (rowCount > 0) {
+                logger.info("Successfully deleted node where group_id : " + heartbeat.getGroupId() + " and node_id : "
+                        + heartbeat.getNodeId() + ".");
+                deleteAllNodeData(productName, heartbeat);
+            } else {
+                throw new DashboardServerException("Error occurred while deleting node where group_id : "
+                        + heartbeat.getGroupId() + " and node_id : " + heartbeat.getNodeId()
+                        + ".");
+            }
+        } catch (DashboardServerException e) {
             throw new DashboardServerException("Error occurred while deleting node where group_id : "
-                                               + heartbeat.getGroupId() + " and node_id : " + heartbeat.getNodeId()
-                                               + ".");
+                    + heartbeat.getGroupId() + " and node_id : " + heartbeat.getNodeId() + ".", e.getCause());
         }
     }
 
     private void deleteAllNodeData(String productName, HeartbeatObject heartbeat) {
-        logger.info("Deleting all information in node : " + heartbeat.getNodeId() + " in group: "
-                 + heartbeat.getGroupId());
-        ArtifactsManager artifactsManager = getArtifactManager(productName, heartbeat);
-        artifactsManager.runDeleteAllExecutorService();
+        try {
+            logger.info("Deleting all information in node : " + heartbeat.getNodeId() + " in group: "
+                    + heartbeat.getGroupId());
+            ArtifactsManager artifactsManager = getArtifactManager(productName, heartbeat);
+            artifactsManager.runDeleteAllExecutorService();
+        } catch (DashboardServerException e) {
+            throw new DashboardServerException("Error occurred while deleting all the nodes: " + e.getMessage(),
```

Comment:
Why wrapping using the same kind of exception?

Reply:
Removed.